### ISSUE TITLE
Freezes or crashes after multiple instancing on aaphostsample

### ISSUE THREAD

**Issue by atsushieno** [atsushieno/android-audio-plugin-framework#68]
It is a typical but intermittent problem that applying a plugin on aaphostsample results in freezes. It most likely happens after 3~4 times on different plugins with LV2 plugins, or crashes with JUCE plugins.

They may be different issues, and more likely plugin-side issues, but at this state it is looking like an overall framework or host problem. JUCE AudioPluginHost shows similar but different unstability too.

**Comment by atsushieno** [comment]
A minor fact: it does not happen with aapbarebonepluginsample.

**Comment by atsushieno** [comment]
Not reproducible with ayumi-lv2 either.

**Comment by atsushieno** [comment]
aap-fluidsynth reproduces the issue, but with androidaudioplugin-lv2 module, instead of aar, included in the aap-lv2-fluidsynth project, whenever I put breakpoints in AAP functions in android-audio-plugin-lv2-bridge.cpp, it just works. There are likely some threading issues.

**Comment by atsushieno** [comment]
Tried the ^ fix on aap-lv2-mda, but no such luck there.

**Comment by atsushieno** [comment]
Again, the fix ^ did not work in aap-sfizz either. But this time there are some suspicious log outputs from sfizz:

> 2021-02-23 18:49:48.345 5709-5841/org.androidaudioplugin.samples.aap_sfizz I/AAPHostNative: [sfizz] Got an unknown atom in work
> 2021-02-23 18:49:48.345 5709-5841/org.androidaudioplugin.samples.aap_sfizz I/AAPHostNative: URI: (null)

**Comment by atsushieno** [comment]
aap-guitarix either. There is another interesting log output there:

> 2021-02-23 19:01:36.561 6423-6433/org.androidaudioplugin.aaphostsample I/n.aaphostsampl: Thread[6,tid=6433,WaitingInMainSignalCatcherLoop,Thread*=0xddf46210,peer=0x13e80268,"Signal Catcher"]: reacting to signal 3

**Comment by atsushieno** [comment]
Maybe ^ was due to general ANR, not much information within itself... https://stackoverflow.com/questions/11302196/why-am-i-getting-threadid-3-reacting-to-signal-3-and-game-freeze-andengine

**Comment by atsushieno** [comment]
None of above actually worked generally, failing on another Android emulator (confirmed with API level 29 / x86).

**Comment by atsushieno** [comment]
It is actually more complicated. aap-mda-lv2 runs fine if I build it with changes to `settings.gradle` and `aap-mda-lv2/build.gradle` to build androidaudioplugin-lv2 within the project (confirmed this by running without debugger). Chances are that release builds don't work - locally published Maven artifacts are release builds.

**Comment by atsushieno** [comment]
It is indeed most likely due to release builds. We'll see if making changes on maven publisher settings works here.

**Comment by atsushieno** [comment]
Now LV2 part is fixed. aap-juce has similar issues. It might be only about libandroidaudioplugin.so, but we'll see...

**Comment by atsushieno** [comment]
JUCE plugins don't freeze like before, but still crashes on occasion. Here is the latest suspicious log lines:

```
2021-02-24 08:32:06.842 8075-8094/com.datsounds.obxd A/.datsounds.obx: java_vm_ext.cc:577] JNI DETECTED ERROR IN APPLICATION: JNI FindClass called with pending exception java.lang.RuntimeException: Only one Looper may be created per thread
    java_vm_ext.cc:577]   at void android.os.Looper.prepare(boolean) (Looper.java:109)
    java_vm_ext.cc:577]   at void android.os.Looper.prepare() (Looper.java:104)
    java_vm_ext.cc:577] 
    java_vm_ext.cc:577]     in call to FindClass
```

**Comment by atsushieno** [comment]
The issue ^ should be fixed at aap-juce@2c043cf.

**Comment by atsushieno** [comment]
This is not an issue anymore, at least as an overall aap core issue.
